Task50/UnitTest1.cs

```diff
@@ -0,0 +1,209 @@
+using OpenQA.Selenium.Support.UI;
+using System;
+using System.Collections.Generic;
+
+namespace Task50
+{
+    public class Tests
+    {
+        IWebDriver _driver;
+
+        //explicit waiter for the webelement
+        bool wait(IWebDriver driver, By locator)
+        {
+            WebDriverWait waiter = new WebDriverWait(_driver, TimeSpan.FromSeconds(10));
+            waiter.PollingInterval = TimeSpan.FromMilliseconds(100); //changing polling frequency
+
+            try
+            {
+                var element = waiter.Until(condition =>
+                {
+                    try
+                    {
+                        var elementToBeDisplayed = _driver.FindElement(locator);
+                        return elementToBeDisplayed.Displayed;
+                    }
+                    catch (NoSuchElementException)
+                    {
+                        return false;
+                    }
+                    catch (StaleElementReferenceException)
+                    {
+                        return false;
+                    }
+
+                });
+
+                return true;
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return false;
+            }
+        }
+
+        // making screenshot, saving to the file in jpeg format
+        void MakeScreenShot(IWebDriver driver, string path)
+        {
+            Screenshot img = ((ITakesScreenshot)driver).GetScreenshot();
+            img.SaveAsFile(path, ScreenshotImageFormat.Png);
+        }
+
+        [SetUp]
+        public void Setup()
+        {
+            _driver = new ChromeDriver();
+        }
+
+        [TearDown]
+        public void CleanUp()
+        {
+            _driver.Close();
+        }
+
+        [TestCase("seleniumtests@tut.by", "123456789zxcvbn")]
+        [TestCase("seleniumtests2@tut.by", "123456789zxcvbn")]
+
+        [Test]
+        public void LoginTest(string loginstr, string passwordstr)
+        {
+            _driver.Navigate().GoToUrl("https://mail.tut.by");
+
+            MakeScreenShot(_driver, "/Users/macbookpro/Desktop/TestScreenshot.png"); //making screenshot, saving it to the Desktop
+
+            IWebElement loginField = _driver.FindElement(By.Id("Username"));
+            IWebElement passwrdField = _driver.FindElement(By.Id("Password"));
+            IWebElement loginButton = _driver.FindElement(By.XPath("//input[contains(@class, \"loginButton\")]"));
+
+            loginField.SendKeys(loginstr);
+            passwrdField.SendKeys(passwordstr);
+            loginButton.Click();
+
+         //   Thread.Sleep(10000); //neither explicit, nor implicit waiter. Implicit waiters work for all the elements. And the explicit waiters works for a particular element with the particular condition. Thread.Sleep() just stop the thread with executing code for a particular time.
+
+           
+
+
+            Assert.IsTrue(wait(_driver, By.XPath("//div[@class=\"mail-User-Name\"]")));
```

Comment:
Please, put this xpath expression outside of assert method. Use variables on the test/class level. Theoretically, these web elements could be used in several tests. And DON'T FORGET to add info messages to assert methods.